fuzz/json_pointer_fuzzer.cc

```diff
@@ -0,0 +1,59 @@
+#include <fuzzer/FuzzedDataProvider.h>
+
+#include "json.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+	FuzzedDataProvider fdp(data, size);
+
+	struct json_tokener *tokener =
+	    json_tokener_new_ex(fdp.ConsumeIntegralInRange<int>(1, JSON_TOKENER_DEFAULT_DEPTH));
+	int flags = 0;
+	if (fdp.ConsumeBool()) {
+		flags |= JSON_TOKENER_VALIDATE_UTF8;
+	}
+	if (fdp.ConsumeBool()) {
+		flags |= JSON_TOKENER_ALLOW_TRAILING_CHARS;
+	}
+	if (fdp.ConsumeBool()) {
+		flags |= JSON_TOKENER_STRICT;
+	}
+	json_tokener_set_flags(tokener, flags);
+
+	std::string path = fdp.ConsumeRandomLengthString(5);
+	std::string sub_json_str = fdp.ConsumeRandomLengthString(10);
+	bool use_format_string = fdp.ConsumeBool();
+	std::string json_str = fdp.ConsumeRemainingBytesAsString();
+
+	struct json_object *jo1 = json_tokener_parse_ex(tokener, json_str.c_str(), json_str.size());
+	if (jo1 == NULL) {
+		json_tokener_free(tokener);
+		return 0;
+	}
+
+	struct json_object *sub_json = json_tokener_parse(sub_json_str.c_str());
+	if (sub_json == NULL) {
+		json_object_put(jo1);
+		json_tokener_free(tokener);
+		return 0;
```

Comment:
Does it make sense to create an empty object, e.g., by calling `json_object_new_object`, instead of returning early if the parsing of the substring fails?

Reply:
Yes, good point. I also removed two of the `if (jso == NULL) return` checks because all the API functions should handle `NULL` values.